usl_models/usl_models/flood_ml/model.py

```diff
@@ -37,21 +32,41 @@ class Input(TypedDict):
 
     def __init__(
         self,
-        model_params: FloodModelParams,
+        params: FloodModelParams | None,
         spatial_dims: tuple[int, int] = (constants.MAP_HEIGHT, constants.MAP_WIDTH),
     ):
         """Creates the flood model.
 
         Args:
-            model_params: A dictionary of configurable model parameters.
+            params: A dictionary of configurable model parameters.
             spatial_dims: Tuple of spatial height and width input dimensions.
                 Needed for defining input shapes. This is an optional arg that
                 can be changed (primarily for testing/debugging).
+            artifact_uri: Optional artifact URI to load model weights from.
         """
-        self._model_params = model_params
+        self._model_params = params or model_params.default_params()
         self._spatial_dims = spatial_dims
         self._model = self._build_model()
 
+    @classmethod
+    def from_checkpoint(cls, artifact_uri: str, **kwargs) -> "FloodModel":
+        """Loads the model from a checkpoint URI.
+
+        We load weights only to keep custom methods (e.g. `call_n`) intact.
+        This only works if the model architecture is identical to the architecure
+        used during export.
+        Ideally, we would load the entire Keras model and use that directly to allow
+        loading different architectures within the same wrapper class.
+        Unfortunately, `call_n` is not trivially serializeable in its current state.
+        """
+        model = cls(**kwargs)
+        loaded_model = keras.models.load_model(os.path.join(artifact_uri, "model"))
+        if loaded_model is not None:
+            model._model.set_weights(loaded_model.get_weights())
+        else:
+            logging.error(f"Failed to load model from: {artifact_uri}")
```

Comment:
should we raise an exception here?
this looks like a failure we can't really recover from

Reply:
Done.